src/backends/cg.rs

```diff
@@ -45,24 +53,31 @@ impl<D: HasDisplayHandle, W: HasWindowHandle> SurfaceInterface<D, W> for CGImpl<
             RawWindowHandle::AppKit(handle) => handle,
             _ => return Err(InitError::Unsupported(window_src)),
         };
-        let view = handle.ns_view.as_ptr() as id;
-        let window: id = unsafe { msg_send![view, window] };
-        let window: id = unsafe { msg_send![window, retain] };
+
+        let mtm =
+            MainThreadMarker::new().expect("can only access AppKit handles on the main thread");
+        let view = handle.ns_view.as_ptr();
+        // SAFETY: The pointer came from `WindowHandle`, which ensures that
+        // the `AppKitWindowHandle` contains a valid pointer to an `NSView`.
+        // Unwrap is fine, since the pointer came from `NonNull`.
+        let view: Id<NSView> = unsafe { Id::retain(view.cast()) }.unwrap();
         let layer = CALayer::new();
+        let subview = unsafe { NSView::initWithFrame(mtm.alloc(), view.frame()) };
+        layer.setContentsGravity(unsafe { kCAGravityTopLeft });
+        layer.setNeedsDisplayOnBoundsChange(false);
+        set_layer(&subview, &layer);
         unsafe {
-            let subview: id = NSView::alloc(nil).initWithFrame_(NSView::frame(view));
-            layer.set_contents_gravity(ContentsGravity::TopLeft);
-            layer.set_needs_display_on_bounds_change(false);
-            subview.setLayer(layer.id());
-            subview.setAutoresizingMask_(NSViewWidthSizable | NSViewHeightSizable);
-
-            view.addSubview_(subview); // retains subview (+1) = 2
-            let _: () = msg_send![subview, release]; // releases subview (-1) = 1
-        }
+            subview.setAutoresizingMask(NSAutoresizingMaskOptions(
+                NSAutoresizingMaskOptions::NSViewWidthSizable.0
+                    | NSAutoresizingMaskOptions::NSViewHeightSizable.0,
+            ))
+        };
+
+        unsafe { view.addSubview(&subview) };
         let color_space = CGColorSpace::create_device_rgb();
         Ok(Self {
             layer,
-            window,
+            window: view.window().expect("view to be inside a window"),
```

Comment:
Not sure if you want to handle this as a panic or as something else?

Reply:
I would return a `BackendError` here as well